ui/app/components/generate-credentials-totp.js

```diff
@@ -4,46 +4,66 @@
  */
 
 import Component from '@glimmer/component';
-import { action } from '@ember/object';
 import { tracked } from '@glimmer/tracking';
-import { later, cancel } from '@ember/runloop';
-import timestamp from 'core/utils/timestamp';
-import { getUnixTime } from 'date-fns';
+import { task, timeout } from 'ember-concurrency';
+import { service } from '@ember/service';
 
 export default class GenerateCredentialsTotp extends Component {
+  @tracked elapsedTime = 0;
+  @tracked totpCode = null;
+  @service store;
+
   title = 'Generate TOTP code';
 
-  @tracked elapsedTime = 0;
-  nextTick = null;
+  constructor() {
+    super(...arguments);
+    const { backendPath, keyName } = this.args;
+    this.generateTotpCode(backendPath, keyName);
```

Comment:
I know you have a todo for this, so this can be addressed later, but it seems like `this.generateTotpCode` should be called in the `startTimer` task so that it requests a new code whenever the task restarts